Asks V1

juniusfree opened the pull request:
This is my first contribution. Please let me know if there's any problem with this PR. Thank you!

Notes:
1. This is the documentation of the contract: https://docs.zora.co/docs/smart-contracts/modules/Asks/zora-v3-asks-intro
2. `setRoyaltyEngineAddress` doesn't have a test tx hash because there's none.

leojacob1 commented:
@juniusfree,

A couple notes...
1. The `fillAsk` function is actually a full NFT sale (https://etherscan.io/tx/0xaab2e7717aa80d1163415a9a43f031dfd117ff21a180b640d11c7bdaec05698a). Please see the contributor doc for how to handle this case (use `"__NFTSALE__"` and `multiSidedTxMap`)
2. The variable that you have to use to get the user's ENS/address in the exampleDescriptionTemplate is `userName` not `username`. Small, I know :)
3. I don't see the `setRoyaltyEngineAddress` method in bloxy (https://bloxy.info/address/0x6170B3C3A54C3d8c854934cBC314eD479b2B29A3). Where did you get that from?
4. The `_handleRoyaltyEnginePayout` looks to be a purely internal method so you didn't have to include that one (https://bloxy.info/txs/calls_sc/0x6170b3c3a54c3d8c854934cbc314ed479b2b29a3?signature_id=4108120).

Great work on this! You can go ahead and push to this PR and I'll review.

Best,
Leo

juniusfree commented:
Thanks @leojacob1!

1. Updated. Please let me know if there's still a problem.
2. Updated.
3. This is included in the JSON generated by EVM translator. I removed it now.
4. Removed.

Please let me know if there's any problem. Thanks again.

leojacob1 commented:
That 0x6170 address is actually the contract address in `multiSidedTxMap` and not an address participating in the tx. (https://etherscan.io/tx/0xaab2e7717aa80d1163415a9a43f031dfd117ff21a180b640d11c7bdaec05698a)

juniusfree commented:
@leojacob1 Sorry about that. I updated it now.
